src/minio/managers/policy_manager.py

```diff
@@ -295,6 +296,114 @@ async def _attach_detach_policy(
                 f"{'Attached' if attach else 'Detached'} policy {policy_name} {'to' if attach else 'from'} {target_type.value} {target_name}"
             )
 
+    # === POLICY DOCUMENT MANIPULATION ===
+
+    def add_path_access_to_policy(
+        self,
+        policy_model: PolicyModel,
+        path: str,
+        permission_level: PolicyPermissionLevel,
+    ) -> None:
+        """
+        Add access permissions for a specific path to an existing policy model.
+
+        Args:
+            policy_model: The policy model to modify (modified in-place)
+            path: The S3 path to grant access to (e.g., "s3a://bucket/path/to/data")
+            permission_level: The level of access to grant (READ, WRITE, or ADMIN)
+
+        Note:
+            This method only modifies the policy model in memory. Call update_policy()
+            to persist the changes to MinIO.
+        """
```

Comment:
This sounds like a race condition. If two people are updating the policy at the same time, one set of changes will get blown away IIUC